tests/e2e/vscode-test-playwright/index.ts

```diff
@@ -0,0 +1,435 @@
+      const cachePath = await _createTempDir();
+      const installBasePath = path.join(
+        process.cwd(),
+        ".vscode-test",
+        `worker-${workerInfo.workerIndex}`,
+      );
+      await fs.promises.mkdir(installBasePath, { recursive: true });
+      const installPath = await downloadAndUnzipVSCode({
+        cachePath: installBasePath,
+        version: vscodeVersion,
+      });
+      const [cliPath] = resolveCliArgsFromVSCodeExecutablePath(installPath);
+
+      if (extensions) {
+        await new Promise<void>((resolve, reject) => {
+          extensions = typeof extensions === "string" ? [extensions] : extensions ?? [];
+          const subProcess = cp.spawn(
+            cliPath,
+            [
+              `--extensions-dir=${path.join(cachePath, "extensions")}`,
+              `--user-data-dir=${path.join(cachePath, "user-data")}`,
+              ...extensions.flatMap((extension) => ["--install-extension", extension]),
+            ],
+            {
+              stdio: "inherit",
+              shell: os.platform() === "win32",
+            },
+          );
+          subProcess.on("exit", (code, signal) => {
+            if (!code) resolve();
+            else
+              reject(new Error(`Failed to install extensions: code = ${code}, signal = ${signal}`));
+          });
+        });
+      }
+
+      await use({ installPath, cachePath });
+    },
+    { timeout: 0, scope: "worker" },
+  ],
+
+  // based on https://github.com/microsoft/playwright-vscode/blob/1d855b9a7aeca783223a7a9f8e3b01efbe8e16f2/tests-integration/tests/baseTest.ts
+  electronApp: [
+    async (
+      { extensionDevelopmentPath, baseDir, _vscodeInstall, vscodeTrace, trace },
+      use,
+      testInfo,
+    ) => {
+      const { installPath, cachePath } = _vscodeInstall;
+
+      // remove all VSCODE_* environment variables, otherwise it fails to load custom webviews with the following error:
+      // InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state
+      const env = { ...process.env } as Record<string, string>;
+      for (const prop in env) {
+        if (/^VSCODE_/i.test(prop)) delete env[prop];
+      }
+
+      console.log("[DEBUG] Launching Electron app...");
+      const electronApp = await _electron.launch({
+        executablePath: installPath,
+        env,
+        args: [
+          // Stolen from https://github.com/microsoft/vscode-test/blob/0ec222ef170e102244569064a12898fb203e5bb7/lib/runTest.ts#L126-L160
+          // https://github.com/microsoft/vscode/issues/84238
+          "--no-sandbox",
+          // https://github.com/microsoft/vscode-test/issues/221
+          "--disable-gpu-sandbox",
+          // https://github.com/microsoft/vscode-test/issues/120
+          "--disable-updates",
+          "--skip-welcome",
+          "--skip-release-notes",
+          "--disable-telemetry",
+          "--use-inmemory-secretstorage",
```

Comment:
This launch option `--use-inmemory-secretstorage` makes the test environment setup much easier since we wouldn't have to configure a Linux key ring (or whatever OS-specific key store).